airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -55,22 +57,17 @@
 from airflow.providers.amazon.aws.utils.tags import format_tags
 from airflow.utils.helpers import chunks
 
-if TYPE_CHECKING:
-    from mypy_boto3_s3.service_resource import Bucket as S3Bucket, Object as S3ResourceObject
-
-T = TypeVar("T", bound=Callable)
-
 logger = logging.getLogger(__name__)
 
 
-def provide_bucket_name(func: T) -> T:
+def provide_bucket_name(func: Callable) -> Callable:
```

Comment:
Is this change relate to this PR?

Reply:
It's minor cleanup - isn't the base definition more of a Java thing? It's not required for the changes to resources.

Reply:
Yeah, I'm with @ellisms on this one.  That doesn't feel very pythonic to me the way it was.

Reply:
Ah, then ok